llvm/test/CodeGen/AIE/aie2/schedule/pre_ra/add2d_inner.mir

```diff
@@ -0,0 +1,125 @@
+# NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py
+#
+# This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# (c) Copyright 2024 Advanced Micro Devices, Inc. or its affiliates
+# RUN: llc -march=aie2 -run-pass=machine-scheduler %s -o - | FileCheck %s
+
+
+# This represents the innermost loop of Conv2D after SW pipelining.
+# We should see most of the VLDA.UPS instructions move down in the loop
+# BB to reduce the reg pressure and avoid spills. They can later be moved back
+# up by the post-RA scheduler. This should also make the 4 acc1024 COPY
+# instructions coalesce-able.
```

Comment:
Could we have a presched + RA example that actually demonstrates reduced spilling?

Reply:
I really tried to manipulate that example, but can't get it to spill without writing absolutely ugly code. I'd leave it like this if that's fine to you. In a follow-up PR where I change the MachinePipeliner, i'll add an end-to-end IR test that shows Add2D getting nicely SW pipelined.